cmd/config-manager/main.go

```diff
@@ -180,22 +180,30 @@ func detectRuntime() (string, *dbus.Conn, error) {
 	// (either containerd or CRI-O).If more than one container runtime systemd unit is found
 	// to be in an active state, the process fails.
 	log.Infof("looking for active runtime units on D-Bus...")
-	units, err := conn.ListUnitsByPatternsContext(context.Background(), []string{"active"}, []string{containerdUnit, crioUnit})
+	units, err := conn.ListUnitsFilteredContext(context.Background(), []string{"active"})
+
+	expectedUnits := make([]dbus.UnitStatus, 0)
+	for _, unit := range units {
+		if unit.Name == "containerd.service" || unit.Name == "crio.service" {
+			expectedUnits = append(expectedUnits, unit)
+		}
+	}
+
 	if err != nil {
```

Comment:
probably error handling would be better before `make()`

Reply:
+1

Reply:
Ops, I accidentally moved already existing error handling down. But I have now fixed it.